app/models/user.rb

```diff
@@ -111,6 +111,11 @@ def self.lookup_by_lower_email(email, cache = [])
     Array.wrap(cache).detect { |u| u.email.try(:downcase) == email } || find_by(['lower(email) = ?', email])
   end
 
+  def lower_userid
+    userid&.downcase
+  end
+
+  virtual_attribute :lower_userid, :string, :arel => ->(t) { t.grouping(t[:userid].lower) }
```

Comment:
Only thing I don't like about this is that it's now exposed to reporting.

Reply:
we need to come up with a way to hide attributes from reporting and from the api
This is also useful when we want to deprecate a fields and have them work, but not show up in the drop downs/documentation

this would add some meta-data but may be a viable solution.
I have had an idea mulling in my head for a while, but never know if this is viable or not - due to the complexity of add ing a concept that isn't really in rails at this time

Reply:
I like the public/private designation idea.  What if we just literally make them private, then in reporting, as we enumerate the methods we only take public virtual methods?

Reply:
If we are using `send` to call these parameters, then that will require changes in using them
but when we are constructing the api or the drop down, we can look at them.

Nice to not introduce any new metadata

I do wonder how this will work if we use the class when it hasn't been used yet. the attributes may not be defined yet.

will look into this